### PR TITLE
PenCode added to News Articles

### PR DESCRIPTION
**Pull request by Deantwo**
You can now use [PenCode](https://tgstation13.org/wiki/Paper_BBCode) in news articles.

Also inclues:
- Multi-line text input for writing news articles.
- The input box now contain the exsiting message as default. No more writing the whole thing again just to fix a small typo.
- Cancelling the article input box no longer deletes the entered article.

Writing:
writing the news

Reading:
reading the news
(NewsPaper on the left)

### CODE REVIEW

**Comment by AndroidSFV** [comment]
Finally someone has done this. Thank you

**Comment by Deantwo** [comment]
And it was an easy thing to implement too.
I never understood why it didn't use pencode from the begining.

**Comment by Deantwo** [comment]
Right now the [sign] uses the same font as the rest of the news article.
Should I change the default article font to the same as paper?

**Comment by pudl** [comment]
make it comic sans

**Comment by Deantwo** [comment]
> make it comic sans

I don't know that font.
And in light of #7263 I don't feel like using fonts that I aren't sure of.
Using the same font as paper as used for years sound safer.

**Comment by RemieRichards** [comment]
@Deantwo you're lucky not to know that font, haha. that was a joke as well, just ignore them. :)

**Comment by Deantwo** [comment]
I think I'll change the font to that of paper as I stated.
But may have to be tomorrow, because it is late and I need sleep.

**Closed until then!**

**Comment by Deantwo** [comment]
Accidentally didn't go to bed when I was suppose to and made the commit.
So hope people will use the NewsCasters more now! I know I will.

File: code/game/machinery/newscaster.dm [hunk 1]
```diff
@@ -165,6 +165,8 @@ var/list/obj/machinery/newscaster/allCasters = list() //Global list that will co
 	var/allow_comments = 1
 	luminosity = 0
 	anchored = 1
+	var/const/articlefont = "Verdana"
+	var/const/signfont = "Times New Roman"
```
Comment (Miauw62) [inline code/game/machinery/newscaster.dm]: Should probably be a define, as such:
#define ARTICLEFONT "Verdana"
#define SIGNFONT "Times New Roman"

File: code/game/machinery/newscaster.dm [hunk 2]
```diff
@@ -873,6 +878,34 @@ var/list/obj/machinery/newscaster/allCasters = list() //Global list that will co
 		qdel(P)
 
 
+/obj/machinery/newscaster/proc/parsepencode(t, mob/user)
```
Comment (Miauw62) [inline code/game/machinery/newscaster.dm]: This is copy-pasted, which is bad because it means that changing the code here means it has to be changed everywhere else too. just make it a global proc. (/proc/parsepencode()) and use that everywhere.

File: code/game/machinery/newscaster.dm (hunk 1 above)
Comment (Deantwo) [inline code/game/machinery/newscaster.dm]: I just did the same as it was done in [Paper.dm](https://github.com/tgstation/-tg-station/blob/bff2e62a0f28e7838ad19f1b955e0f4250362e02/code/modules/paperwork/paper.dm).
I change it there too, but would like to do it in a fixing PR for after [both my PRs](https://github.com/tgstation/-tg-station/pull/7287) are merged.
Need to do as I [discussed with RemieRichards](https://github.com/tgstation/-tg-station/pull/7287#issuecomment-71113858).

File: code/game/machinery/newscaster.dm (hunk 2 above)
Comment (Deantwo) [inline code/game/machinery/newscaster.dm]: Read what I [discussed with RemieRichards](https://github.com/tgstation/-tg-station/pull/7287#issuecomment-71113858).
TL;DR: I'll fix it once they are merged.

Comment (Miauw62) [inline code/game/machinery/newscaster.dm]: w/e. @RemieRichards it's your job to keep track of this and ensure that it happens V:
